File: cpp/tests/bybit_l2_variants.cpp
#include "llab/bybit_l2_dense_ladder.hpp"
#include "llab/bybit_l2_ordered_map.hpp"
#include <stdexcept>
int main(){using namespace llab::bybit_l2; std::vector<Level>b{{100,2},{99,1}},a{{101,3}};std::vector<Change>d{{Side::Bid,100,1},{Side::Ask,101,0},{Side::Ask,102,4}};DenseLadderBook dense(90,20);OrderedMapBook tree;if(!dense.apply_snapshot(1,b,a)||!tree.apply_snapshot(1,b,a)||!dense.apply_delta(2,d)||!tree.apply_delta(2,d)||dense.best_bid()!=tree.best_bid()||dense.best_ask()!=tree.best_ask())throw std::runtime_error("variant parity failed");}

File: rust/tests/bybit_l2_variants.rs
use low_latency_lab_benchmarks::bybit_l2_order_book::{Change, Level, Side};
use low_latency_lab_benchmarks::bybit_l2_dense_ladder::DenseLadderBook;
use low_latency_lab_benchmarks::bybit_l2_ordered_map::OrderedMapBook;
#[test] fn dense_and_tree_match_vector_semantics() {
 let bids=[Level{price:100,quantity:2},Level{price:99,quantity:1}];let asks=[Level{price:101,quantity:3}];let d=[Change{side:Side::Bid,price:100,quantity:1},Change{side:Side::Ask,price:101,quantity:0},Change{side:Side::Ask,price:102,quantity:4}];
 let mut dense=DenseLadderBook::new(90,20);let mut tree=OrderedMapBook::new();assert!(dense.apply_snapshot(1,&bids,&asks)&&tree.apply_snapshot(1,&bids,&asks));assert!(dense.apply_delta(2,&d)&&tree.apply_delta(2,&d));assert_eq!(dense.best_bid(),tree.best_bid());assert_eq!(dense.best_ask(),tree.best_ask());
}
